extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java

```diff
@@ -128,6 +133,50 @@ public class InfinispanClientRuntimeConfig {
     @ConfigItem
     Optional<String> trustStoreType;
 
+    /**
+     * Creates the caches on first access with the provided configuration.
+     */
+    @ConfigItem
+    Map<String, RemoteCacheConfig> cache = new HashMap<>();
+
+    @ConfigGroup
+    public static class RemoteCacheConfig {
+
+        /**
+         * Provides the configuration in XML
+         */
+        @ConfigItem
+        public String configuration;
```

Comment:
TBH, this looks very very weird. Why it is needed?

Reply:
@geoand to pass configuration of the caches in the properties and create caches from the client on first access. This is how Infinispan works https://github.com/infinispan-demos/harry-potter-quarkus/blob/main/src/main/resources/META-INF/hotrod-client.properties

Reply:
I understand that, but it's super weird (not to mention very user unfriendly) to add XML in a value of a properties file... 
Is this property really needed?

Reply:
My point is that unless we really need this, I would much like to have this removed

Reply:
We are experiencing issues related to the fact of spinning up a server and creating a cache from the client that leads to race conditions. plus the fact of willing a cache for demo or dev mode only purpose, and not in production.
The way to fix those issues is providing the configuration this way. Another additional option is also providing a configuration URI.

Reply:
Is there no other way to configure this? I mean, can't the client configure the stuff that XML describes?

I'll also loop @Sanne in on this, since he is in a unique position to know a lot about Infinispan and Quarkus :)

Reply:
No, we can't create that from the client using properties. We can configure the client properties concerning a cache for the client (near caching, marshaller...) but the cache has its own configuration and lives in the server. Caches usually exist or are created by users in production with the operator for example. However, when we talk about spinning up a demo or having caches in test mode or dev mode, the server won't have those caches. We can use the administration API as we do today in the quarkus quickstart to create the cache. This leads to race condition (the remotecache is injected before the cache is created).

Reply:
Understood, thanks

Reply:
however, we have built a new feature in the Console to help users configure and/or create caches in Infinispan 14.
Some screenshots here: https://github.com/infinispan/infinispan.github.io/pull/302/files

Reply:
So you're literally expecting to inline XML into the properties file? I agree that looks weird :)

But also, people would most likely prefer to edit XML files in an editor, to get autocompletion and validations. The examples below show a very simple configuration, but how does that scale realistically?

Wouldn't it be better if it was referencing a resource name of a separate XML resource?

TBH I even assumed that was the case initially - if you have strong reasons to prefer this inlined, then I think the description of the configuration property needs to be clarified.

Reply:
I don't have any personal strong reason to prefer this inlined, is the way it works in the hotrod-client.properties 

https://infinispan.org/docs/stable/titles/hotrod_java/hotrod_java.html#creating-caches-hotrod_hotrod-client-configuration

I'm more than opened to improve it for quarkus use case of course

Reply:
@Sanne concerning the scaling question, yes, it does not scale, but it's not meant to be used that way neither in infinispan :) we have configuration_uri as well that can be used instead of pasting the config that is the most simple way to do it now! :)

Reply:
ic. Personally I think it's weird what the hot rod client did, but I guess it sets a precedent :)

I guess it had reasons. In Quarkus we generally also strive to have it all in a single configuration file so I guess it's not too bad - it's a bit odd that it's XML but I guess it's not a blocker.

On the upside, this would make it easier for people to inject a new configuration using environment variables, which is valuable in Kubernetes.

So I guess it's Ok with me :+1:

Reply:
Yes, of course. I can change the documentation description without trouble. I can also implement the URI configuration and test if it works well. With the new console feature, we can create pretty good formed cache configuration and get it in yaml, xml or json as needed. This should be a big usability feature in Infinispan caches configuration too.